app/components/avo/resource_component.rb

```diff
@@ -20,7 +20,14 @@ def can_delete?
   end
 
   def can_detach?
-    authorize_association_for(:detach)
+    return false if @reflection.blank? || @resource.model.blank? || !authorize_association_for(:detach)
+
+    # If the inverse_of is a belongs_to, we need to check if it's optional in order to know if we can detach it.
+    if @reflection.inverse_of.is_a?(ActiveRecord::Reflection::BelongsToReflection)
```

Comment:
What if it does not have an `inverse_of` configured on the association? Will this fail?
Can we raise a helpful error before, so the user knows about declaring the inverse_of?

Reply:
From my manual tests even if `inverse_of` is not configured rails can find it, not sure if there are cases where this can fail

Reply:
> The automatic guessing of the inverse association uses a heuristic based on the name of the class, so it may not work for all associations, especially the ones with non-standard names.

From https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many